tests/Plugin_Context_Tests.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Tests for the Plugin_Context class.
+ *
+ * @package plugin-check
+ */
+
+use WordPress\Plugin_Check\Plugin_Context;
+
+class Plugin_Context_Tests extends WP_UnitTestCase {
+
+	/**
+	 * Create a single Plugin_Context to test.
+	 */
+	public function set_up() {
+		parent::set_up();
+
+		$this->plugin_context = new Plugin_Context( 'plugin-check/plugin-check.php' );
+	}
+
+	/**
+	 * Test class can be instantiated.
+	 *
+	 * @test
+	 */
+	public function it_can_be_instantiated() {
+		$main = new Plugin_Context( 'plugin-check/plugin-check.php' );
+
+		$this->assertIsObject( $main );
+		$this->assertInstanceOf( Plugin_Context::class, $main );
+	}
+
+	/**
+	 * Test basename method.
+	 *
+	 * @test
+	 */
+	public function it_can_return_the_basename() {
```

Comment:
See my comments on #44 and #46 regarding usage of `test_` prefix (applies throughout this class). The convention is to just call the method like the tested method (i.e. here `test_basename()`), unless there are multiple test methods for the same method in which case the method should be more descriptive (e.g. `test_basename_returns_basename()`, though here that is a pointless example).